cmd/genz/genz.go

```diff
@@ -29,7 +29,8 @@ Flags:`
 
 var (
 	generateCmd      = flag.NewFlagSet("", flag.ExitOnError)
-	typeName         = generateCmd.String("type", "", "name of the struct to parse")
+	typeKind         = generateCmd.String("kind", "struct", "kind of type to parse")
```

Comment:
I feel that this is not necessary, types declarations names are unique across a package